x-pack/plugins/security_solution/public/detections/containers/detection_engine/rules/use_rule_from_timeline.tsx

```diff
@@ -103,18 +90,32 @@ export const useRuleFromTimeline = (setRuleQuery: SetRuleQuery): RuleFromTimelin
   );
   // end browser field management
 
+  const getInitialUrlParamValue = useGetInitialUrlParamValue<string>(RULE_FROM_TIMELINE_URL_PARAM);
+  const timelineIdFromUrl = useMemo(getInitialUrlParamValue, [getInitialUrlParamValue]);
+
+  const getInitialUrlParamValueEql = useGetInitialUrlParamValue<string>(RULE_FROM_EQL_URL_PARAM);
+  const timelineIdFromUrlEql = useMemo(getInitialUrlParamValueEql, [getInitialUrlParamValueEql]);
+
   // start set rule
   const handleSetRuleFromTimeline = useCallback(() => {
     if (selectedTimeline == null || selectedDataViewBrowserFields == null) return;
 
-    const newQuery = {
-      query: selectedTimeline.kqlQuery.filterQuery?.kuery?.expression ?? '',
-      language: selectedTimeline.kqlQuery.filterQuery?.kuery?.kind ?? 'kuery',
-    };
-    const newFilters = selectedTimeline.filters ?? [];
+    const isEql = timelineIdFromUrlEql != null;
+    const newQuery = isEql
```

Comment:
This is fine, I think future proofing a bit, you could have an `eqlRuleFromTimeline` helper than generates the `newQuery`, `newFilters`, `dataProvidersDsl`, etc.. In the future there'll probably be a function like this for every rule

Reply:
https://github.com/elastic/kibana/pull/151091/commits/9b1523f5291920cc68d313e8a3159b195812161e